File: packages/cli/tests/fixtures/sample.rb
# Fixture: Ruby with various ENV access patterns

# Standard bracket access
database_url = ENV['DATABASE_URL']
secret_key = ENV["SECRET_KEY_BASE"]

# ENV.fetch with and without default
stripe_key = ENV.fetch('STRIPE_SECRET_KEY')
redis_url = ENV.fetch('REDIS_URL', 'redis://localhost:6379')

# ENV.dig (less common)
maybe_value = ENV.dig('OPTIONAL_VAR')

# In a class
class Config
  def self.stripe_key
    ENV['STRIPE_PUBLISHABLE_KEY']
  end

  def self.admin_email
    ENV.fetch('ADMIN_EMAIL')
  end
end

# In a Rails initializer style
if ENV['RAILS_ENV'] == 'production'
  config.force_ssl = true
end

# This should be skipped — it's commented out
# secret = ENV['COMMENTED_SECRET']

# Dynamic access — flagged as __DYNAMIC__
key = :SOME_SETTING
dynamic = ENV[key]

File: packages/cli/tests/fixtures/sample.py
import os
from os import environ

# Standard os.environ access
db_url = os.environ['DATABASE_URL']
debug = os.environ["DEBUG_MODE"]

# os.environ.get with and without default
stripe_key = os.environ.get('STRIPE_SECRET_KEY')
redis_url = os.environ.get('REDIS_URL', 'redis://localhost:6379')

# os.getenv
secret_key = os.getenv('SECRET_KEY')
jwt_secret = os.getenv('JWT_SECRET', 'dev-secret')

# From imported environ
admin_email = environ['ADMIN_EMAIL']
allowed_hosts = environ.get('ALLOWED_HOSTS', 'localhost')

# In a function
def get_database_config():
    return {
        'host': os.environ.get('DB_HOST', 'localhost'),
        'port': os.environ.get('DB_PORT', '5432'),
        'name': os.environ['DB_NAME'],
    }

# This should be skipped — it's commented out
# password = os.environ['COMMENTED_PASSWORD']

# Dynamic access — flagged as __DYNAMIC__
key_name = 'SOME_VAR'
dynamic = os.environ[key_name]
